ql/src/java/org/apache/hadoop/hive/ql/exec/TaskRunner.java

```diff
@@ -110,10 +109,11 @@ public void runSequential() {
       LOG.error("Error in executeTask", t);
     }
     result.setExitVal(exitVal);
-    taskQueue.releaseRunnable();
     if (tsk.getException() != null) {
       result.setTaskError(tsk.getException());
     }
+    result.setRunning(false);
+    taskQueue.releaseRunnable();
```

Comment:
can we avoid all these {{result.setRunning(false)}} change and just pull {{result.setExitVal(exitVal)}} below {{result.setTaskError(tsk.getException());}}
I tried locally and it worked for me.

Reply:
Done.